### PR TITLE
Add `--jwt-id` flag

### LINKED ISSUE #12988: Feature request: IDs for JWT claims
# 🚀 Feature Request

### Description

The execution API spec defines an optional `id` claim that can be set in a JWT token to convey information about the client being authenticated. This is useful for more advanced forms of authentication beyond the basic 1-CL-to-1-EL authentication that we usually do. For example, my [_Electric Eel_](https://github.com/sigp/eleel) multiplexer uses the `id` to identify the connecting client and decide which JWT secret to use. Without the ID, it has to fall back on iterating all known JWT secrets and trying to verify the token against them (which is slow).

The spec defines the `id` claim here: https://github.com/ethereum/execution-apis/blob/main/src/engine/authentication.md#jwt-claims

It's worth noting that the ID in the claim is _distinct_ from the key-ID concept defined by the JWT standard, which is part of the _header_, not the body. Apart from being a bit inelegant this isn't an issue.

### Describe the solution you'd like

Add a flag to Prysm like `--jwt-id` which takes a string to use as the JWT ID for claims made to the EL.

### Versions

* Software version: Prysm v4.0.8

### PR DESCRIPTION
**Pull request by bnovil**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**
Add an optional flag `--jwt-id` to identify the client

**Which issues(s) does this PR fix?**

Fixes #12988

**Other notes for review**

### CODE REVIEW

**Review by terencechain — COMMENTED** [review]
This looks good to me. Will be good to get another pair 👀 before merging this

File: network/auth_test.go [hunk 1]
```diff
@@ -51,3 +51,91 @@ func TestJWTAuthTransport(t *testing.T) {
...
+		iat, ok := item.(float64)
+		require.Equal(t, true, ok)
+		issuedAt := time.Unix(int64(iat), 0)
+		// The claims should have an "iat" field (issued at) that is at most, 5 seconds ago.
+		since := time.Since(issuedAt)
+		require.Equal(t, true, since <= time.Second*5)
+		// check jwt claims id
+		id, ok := claims["id"]
+		require.Equal(t, true, ok)
+		require.Equal(t, true, id == jwtId)
```
Comment (prestonvanloon) [inline network/auth_test.go:94]:
```suggestion
		require.Equal(t, id, jwtId)
```

**Review by prestonvanloon — COMMENTED** [review]
One small suggestion on the test, otherwise LGTM

File: network/auth_test.go (hunk 1 above)
Comment (bnovil) [reply network/auth_test.go]: Thanks for your suggestion!

**Review by prestonvanloon — DISMISSED** [review]

**Comment by prestonvanloon** [comment]
thanks!

**Review by prestonvanloon — APPROVED** [review]
Fixed failing test. LGTM

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**